application/forms/EntryForm.php

```diff
@@ -311,114 +317,233 @@ protected function assemble()
     public function addEntry(int $scheduleId): void
     {
         $data = $this->formValuesToDb();
+        $membershipHash = $this->getMembershipHashFromRecipients($scheduleId);
         $recipients = array_map(function ($recipient) {
             return explode(':', $recipient, 2);
         }, explode(',', $this->getValue('recipient')));
 
         $db = Database::get();
         $db->beginTransaction();
 
-        $db->insert('timeperiod', ['owned_by_schedule_id' => $scheduleId]);
-        $timeperiodId = $db->lastInsertId();
+        $members = ScheduleMember::on(Database::get())
+            ->with('timeperiod')
+            ->filter(
+                Filter::all(
+                    Filter::equal('timeperiod.owned_by_schedule_id', $scheduleId),
+                    Filter::equal('schedule_member.membership_hash', $membershipHash)
+                )
+            );
+
+        if ($members->count() > 0) {
+            $timeperiodId = $members->first()->timeperiod_id;
+        } else {
+            $db->insert('timeperiod', ['owned_by_schedule_id' => $scheduleId]);
+            $timeperiodId = $db->lastInsertId();
+        }
 
         $db->insert('timeperiod_entry', $data + ['timeperiod_id' => $timeperiodId]);
 
-        foreach ($recipients as list($type, $id)) {
-            if ($type === 'contact') {
-                $db->insert('schedule_member', [
-                    'schedule_id' => $scheduleId,
-                    'timeperiod_id' => $timeperiodId,
-                    'contact_id' => $id
-                ]);
-            } elseif ($type === 'group') {
-                $db->insert('schedule_member', [
-                    'schedule_id' => $scheduleId,
-                    'timeperiod_id' => $timeperiodId,
-                    'contactgroup_id' => $id
-                ]);
+        if ($members->count() === 0) {
+            $binaryMemberHash = (new Binary([]))->toDb($membershipHash, null, null);
```

Comment:
```suggestion
            $q = ScheduleMember::on(Database::get());
            $binaryMemberHash = $q->getResolver()->getBehaviors($q->getModel())->persistProperty($membershipHash, 'membership_hash');
```

This way you don't hardcode the binary behavior here and it will still work once are compatible with MySQL.